File: Tasks/signin/siii/DataLayer/MUserDL.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using siii.BusinessLayer;
using siii.UserInterface;
using System.IO;

namespace siii.DataLayer
{
    class MUserDL
    {
        public static List<MUser> userList = new List<MUser>();

        public static void storeDataInList(MUser user)
        {
            userList.Add(user);
        }

        public static MUser SignIn(MUser m)
        {
            foreach(MUser t in userList)
            {
                if(t.name == m.name && t.password == m.password)
                {
                    return t;
                }
            }
            return null;

        }
        public static void loadUser(string path)
        {
            string line;
            if (File.Exists(path))
            {
                StreamReader read = new StreamReader(path);
                while ((line = read.ReadLine()) != null)
                {
                    string[] userFields = line.Split(',');
                    MUser u1 = new MUser();
                    u1.name = userFields[0];
                    u1.password = userFields[1];
                    u1.role = userFields[2];
                    userList.Add(u1);
                }
                read.Close();
            }

        }

        public static void storeUser(string path,MUser user)
        {
            StreamWriter file = new StreamWriter(path, true);
            file.WriteLine(user.name + "," + user.password + "," + user.role);
            file.Flush();
            file.Close();
        }
    }
}

File: Tasks/Python/MUserDL.py
import os.path
from MUser import MUser

class MUserDL:
    userList = []

    @staticmethod
    def addUserToList(user):
        MUserDL.userList.append(user)

    @staticmethod
    def SignIn(user):
        for i in MUserDL.userList:
            if (i.userName == user.userName and i.userPassword == user.userPassword):
                return i
        return None

    @staticmethod
    def parseData(line):
        line = line.split(",")
        return line[0], line[1], line[2]

    @staticmethod
    def readDataFromFile(path):
        if (os.path.exists(path)):
            fileVariable = open(path, 'r')
            records = fileVariable.read().split("\n")
            fileVariable.close()
            for line in records:
                userName, userPassword, userRole = MUserDL.parseData(line)
                user = MUser(userName, userPassword, userRole)
                MUserDL.addUserToList(user)
            return True
        else:
            return False

    @staticmethod
    def storeUserIntoFile(user, path):
        file = open(path, 'a')
        file.write("\n"+ user.userName + "," + user.userPassword + "," + user.userRole)
        file.close()
